source/NoteRefFilter.cpp

```diff
@@ -0,0 +1,51 @@
+// Header file
+#include "NoteRefFilter.h"
+
+// System libraries
+#include <String.h>
+#include <Entry.h>
+#include <Node.h>
+#include <NodeInfo.h>
+
+// Constructor
+NoteRefFilter :: NoteRefFilter() {
+}
+
+// Filter hook method
+bool NoteRefFilter :: Filter(const entry_ref *ref,
+			BNode *node,
+			struct stat_beos *stat,
+			const char *mimeType) {
+
+	// Follow links
+	BEntry entry(ref,true);
+	if (entry.InitCheck() != B_OK)
+		return false;
+
+	// Return true if the file is a directory
+	if (entry.IsDirectory())
+		return true;
+
+	// Create a BNode and a BNodeInfo to get the MIME type
+	BNode n(&entry);
+	BNodeInfo ninfo(&n);
+
+	// Create a string to hold the MIME type
+	BString m;
```

Comment:
Switched to a cstring but I left the name as `m` because it conflicts with the argument with the same name.
```cpp
bool NoteRefFilter :: Filter(...,
				const char *mimeType); //   <--   conflicts with char mimeType[B_MIME_TYPE_LENGTH];
```